include/caffe/filler.hpp

```diff
@@ -94,6 +94,59 @@ class GaussianFiller : public Filler<Dtype> {
   shared_ptr<SyncedMemory> rand_vec_;
 };
 
+
+/// @brief Fills a Blob with Gaussian-distributed values @f$ x = a @f$.
+//         RNG is only called once, and its output is stored in
+//         GaussianStaticFillerdata. Followup calls to the Filler function
+//         will bypass RNG.
+template <typename Dtype>
+class GaussianStaticFiller : public Filler<Dtype> {
+ public:
+  explicit GaussianStaticFiller(const FillerParameter& param)
+      : Filler<Dtype>(param) {
+        GaussianStaticFillerFlag=false;
+      }
+  virtual void Fill(Blob<Dtype>* blob) {
+    Dtype* data = blob->mutable_cpu_data();
+    CHECK(blob->count());
+
+    if(GaussianStaticFillerFlag==false){
+      GaussianStaticFillerdata=(Dtype*) malloc(blob->count()*sizeof(Dtype));
+      caffe_rng_gaussian<Dtype>(blob->count(), Dtype(this->filler_param_.mean()),
```

Comment:
It's better to rely on type deduce here, so try `caffe_rng_gaussian<Dtype>(...` --> `caffe_rng_gaussian(...`